### PR TITLE
OLS-86: Doc about cache

### PR DESCRIPTION
**Pull request by tisnik**
## Description

- Doc about cache
- Updated ToC

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [x] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)
- [ ] CI configuration change
- [ ] Konflux configuration change

## Related Tickets & Documents

- Related Issue #[OLS-86](https://issues.redhat.com//browse/OLS-86)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -609,6 +622,38 @@ Currently there exist three conversation history cache implementations:
 1. Redis cache
 1. Postgres cache
 
+Entries stored in cache have compound keys that consist of `user_id` and `conversation_id`. It is possible for one user to have multiple conversations and thus multiple `conversation_id` values at the same time. Global cache capacity can be specified. The capacity is measured as number of entries; entries size are ignored in this computation.
```
Comment (JoaoFula) [inline README.md:625]: The capacity is measured as **the** number of entries

File: README.md [hunk 2]
```diff
@@ -609,6 +622,38 @@ Currently there exist three conversation history cache implementations:
...
+
+#### In-memory cache
+
+In-memory cache is implemented as a queue with defined maximum capacity specified as number of entries that can be stored in a cache. That number is limit for all cache entries, not matter how many users are using LLM. When new entry is put into the cache and if the maximum capacity is reached, oldest entry is removed from the cache.
```
Comment (JoaoFula) [inline README.md:629]: as a queue with **a** defined maximum capacity specified as **the** number of entries...

Comment (JoaoFula) [reply README.md:629]: That number is **the** limit for all cache entries

Comment (JoaoFula) [reply README.md:629]: When **the** new entry (...), **the** oldest entry...

File: README.md [hunk 3]
```diff
@@ -609,6 +622,38 @@ Currently there exist three conversation history cache implementations:
...
+
+#### Redis cache
+
+Entries are stored in Redis as dictionary. LRU policy can be specified that allows Redis to automatically remove oldest entries.
```
Comment (JoaoFula) [inline README.md:633]: remove **the** oldest entries.

File: README.md [hunk 4]
```diff
@@ -609,6 +622,38 @@ Currently there exist three conversation history cache implementations:
...
+
+#### Postgres cache
+
+Entries are stored in one Postgres table with following schema:
```
Comment (JoaoFula) [inline README.md:637]: with **the** following schema:

File: README.md [hunk 5]
```diff
@@ -609,6 +622,38 @@ Currently there exist three conversation history cache implementations:
...
+ value           | bytea                       |          |         | extended |
+ updated_at      | timestamp without time zone |          |         | plain    |
+Indexes:
+    "cache_pkey" PRIMARY KEY, btree (user_id, conversation_id)
+    "cache_key_key" UNIQUE CONSTRAINT, btree (key)
+    "timestamps" btree (updated_at)
+Access method: heap
+```
+
+During new record insertion the maximum number of entries is checked and when the defined capacity is reached, oldest entry is deleted.
```
Comment (JoaoFula) [inline README.md:653]: During **a** new record (...), the oldest entry is deleted.

Comment (tisnik) [reply README.md]: fixed, thank you

File: README.md (hunk 4 above)
Comment (tisnik) [reply README.md]: fixed, thank you

File: README.md (hunk 3 above)
Comment (tisnik) [reply README.md]: fixed, thank you

File: README.md (hunk 2 above)
Comment (tisnik) [reply README.md]: fixed, thank you

**Comment by tisnik** [comment]
/override "Red Hat Konflux / lightspeed-service-on-pull-request"

**Pull request merged**